api/v1beta1/disruption_webhook.go

```diff
@@ -626,43 +644,45 @@ func safetyNetCountNotTooLarge(r *Disruption) (bool, string, error) {
 	return false, "", nil
 }
 
-// safetyNetNeitherHostNorPort is the safety net regarding missing host and port values.
-// it will check against all defined hosts in the network disruption spec to see if any of them have a host and a
-// port missing. The more generic a hosts tuple is (Omitting fields such as port), the bigger the blast radius.
-func safetyNetNeitherHostNorPort(r Disruption) bool {
+// safetyNetMissingNetworkFilters is the safety net regarding missing host, service, or port filters.
+// it will check if any filters have been placed to limit which network traffic is disrupted
+func safetyNetMissingNetworkFilters(r Disruption) bool {
 	if r.Spec.Unsafemode != nil && r.Spec.Unsafemode.DisableNeitherHostNorPort {
 		return false
 	}
 
-	// if hosts are not defined, this also falls into the safety net
-	if r.Spec.Network.Hosts == nil || len(r.Spec.Network.Hosts) == 0 {
-		return true
+	if r.Spec.Network == nil {
+		return false
 	}
 
-	for _, host := range r.Spec.Network.Hosts {
-		if host.Port == 0 && host.Host == "" {
-			return true
-		}
+	if len(r.Spec.Network.Services) > 0 {
+		return false
 	}
 
-	return false
+	// if hosts are not defined, this also falls into the safety net
+	if len(r.Spec.Network.Hosts) > 0 {
+		return false
+	}
+
+	// If we are disrupting network traffic, but we have set NO services, and NO hosts, there are no filters
+	return true
 }
 
-// safetyNetAllowRootDiskFailure is the safety net regarding missing path or invalid path values for a disk failure disruption.
-func safetyNetAllowRootDiskFailure(r *Disruption) (bool, string) {
+// safetyNetAttemptsNodeRootDiskFailure is the safety net regarding node-level disk failure disruption that target the entire fs.
+func safetyNetAttemptsNodeRootDiskFailure(r *Disruption) bool {
 	if r.Spec.Unsafemode != nil && r.Spec.Unsafemode.AllowRootDiskFailure {
-		return false, ""
+		return false
 	}
 
 	if r.Spec.Level != chaostypes.DisruptionLevelNode {
-		return false, ""
+		return false
 	}
 
 	for _, path := range r.Spec.DiskFailure.Paths {
 		if strings.TrimSpace(path) == "/" {
-			return true, "the specified path for the disk failure disruption targeting a node must not be \"/\"."
+			return true
 		}
 	}
 
-	return false, ""
+	return false
```

Comment:
This part is not covered by unit test

Reply:
It doesn't because it's not a target level pod.

Reply:
A pod level spec would return false at L678. A node level spec with a valid path returns false here at L687. I see now that we don't test any valid paths in the unit tests. I'll add another test